docs: add channel and revision example and clarification to htg charms (#892)
juju/terraform-provider-juju: docs-rtd/howto/manage-charms.md

}
```

You can also specify a charm channel and revision. For example:

```terraform
resource "juju_application" "this" {
    model = <model>
    charm {
        name     = "<charm-name>"
        channel  = "<channel-name>"
        revision = "<revision-number>"
    }
}
```

- If both `channel` and `revision` are specified (recommended for reproducibility), the Terraform provider will deploy the requested revision. (The channel will be used later when you update the charm, though see [issue 498](https://github.com/juju/terraform-provider-juju/issues/498) and [issue 881](https://github.com/juju/terraform-provider-juju/issues/881).)
- If only `channel` is specified, the provider will deploy the latest revision available in that channel at application creation time. The charm will not be refreshed on subsequent `terraform apply` runs.

> - If only `revision` is specified, the provider will deploy that revision from the `stable` channel.

Reviewer: Some comments on this,
- I thought you had to specify a channel?
- If you don't have to, I guess it's taking the default channel. 
- `stable` is a risk not a full channel like `latest/stable`. So we should clarify that it will use the default channel.
- We should mention where you can find the default channel i.e. you can see it on charmhub.

Reviewer: Nonetheless, stable is just a risk not a full channel - https://snapcraft.io/docs/channels

Reviewer 2: so it will try to deploy from "latest/stable"? Which might not be the default channel

Author: OK, changing back to default channel. Thanks for checking!

Reviewer 2: So "stable" might be a keyword. But what if you have a stable track? I'm very confused

Reviewer 2: they do have a "latest/stable" channel

Reviewer 2: It all makes sense with the snapcraft doc but it caught me surprised.

Author: @SimoneDutto The fact that leaving out the track uses the default channel instead of `latest` is a surprise to me as well. Based on https://documentation.ubuntu.com/juju/3.6/reference/charm/#track  I would have expected the default track to be `latest`. Looks like I need to update that bit there too.

> - If neither field is specified, the provider will deploy the revision that is the latest revision in the default channel (as set for the charm on Charmhub) at application creation time. The charm will not be refreshed on subsequent `terraform apply` runs.

Reviewer: ```suggestion
- If neither field is specified, the provider will deploy the latest revision from the default channel (as set for the charm on Charmhub). The charm will not be refreshed on subsequent `terraform apply` runs.
```

> - If neither field is specified, the provider will deploy the latest revision from the `latest/stable` channel at application creation time. The charm will not be refreshed on subsequent `terraform apply` runs.

Reviewer: It will deploy the latest revision from the default channel, which may not be `latest/stable`.
